Runtime/Behaviours/CinemachineStoryboard.cs

```diff
@@ -120,12 +139,40 @@ class CanvasInfo
             if (vcam != VirtualCamera || stage != CinemachineCore.Stage.Finalize)
                 return;
 
+            UpdateRenderCanvas();
+
             if (m_ShowImage)
                 state.AddCustomBlendable(new CameraState.CustomBlendable(this, 1));
             if (m_MuteCamera)
                 state.BlendHint |= CameraState.BlendHintValue.NoTransform | CameraState.BlendHintValue.NoLens;
         }
 
+        /// <summary>
+        /// Camera render modes supported by CinemachineStoryboard.
+        /// </summary>
+        public enum StoryboardRenderMode
+        {
+            ScreenSpaceOverlay,
+            ScreenSpaceCamera
+        };
+        
+        StoryboardRenderMode m_PreviousRenderMode;
+        int m_PreviousSortingOrder;
+        float m_PreviousPlaneDistance;
+        void UpdateRenderCanvas()
+        {
+            if (m_PreviousRenderMode != m_RenderMode || 
+                m_PreviousSortingOrder != m_SortingOrder || 
+                Mathf.Abs(m_PreviousPlaneDistance - m_PlaneDistance) > UnityVectorExtensions.Epsilon)
+            {
+                m_PreviousRenderMode = m_RenderMode;
+                m_PreviousSortingOrder = m_SortingOrder;
+                m_PreviousPlaneDistance = m_PlaneDistance;
+                DestroyCanvas();
```

Comment:
You only need to destroy/recreate the canvas if the mode changes.
The other 2 values can just be poked every frame into the existing canvas.  I'm thinking that the way it is now, it's very inefficient to animate the values (someone might do that).